libs/adapters/src/rabbitmq/types.ts

```diff
@@ -77,6 +80,15 @@ export enum RascalRoutingKeys {
   ContestProjectionContestContentAdded = EventNames.ContestContentAdded,
   ContestProjectionContestContentUpvoted = EventNames.ContestContentUpvoted,
 
+  XpProjectionSignUpFlowCompleted = EventNames.SignUpFlowCompleted,
+  XpProjectionCommunityCreated = EventNames.CommunityCreated,
+  XpProjectionCommunityJoined = EventNames.CommunityJoined,
+  XpProjectionThreadCreated = EventNames.ThreadCreated,
+  XpProjectionThreadUpvoted = EventNames.ThreadUpvoted,
```

Comment:
I think this needs to match the `NotificationsProviderThreadUpvoted` routing key i.e. `${EventNames.ThreadUpvoted}.#`. I think the current key will match only events where the routing key is exactly the event. That is, the current key will not route thread upvotes on contest threads where the key is `ThreadUpvoted.contest`

Same goes for `ThreadCreated`

Reply:
We'll discuss routing keys soon, this .contest exception might not be necessary